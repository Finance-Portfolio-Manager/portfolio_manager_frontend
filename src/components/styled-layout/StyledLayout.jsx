/*  WIP
* Component by Team 2 | Carlos Galvan Jr

This is a DEMO for design team, more functionality tbd
*/
import React from "react";
import './layout.css'
import "antd/dist/antd.css";
// import "./index.css";
import { Layout, Menu, Breadcrumb, Card, Col, Row } from "antd";
import { Content, Footer, Header } from "antd/lib/layout/layout";
// import GrandFetchCard from "./chart.js-components/Total/GrandFetchCard";
import FetchCard from "./chart.js-components/Single/FetchCard";
import { ChartVanilla } from "./chart.js-components/vanilla/ChartCanvas";
// import { Chart } from "react-chartjs-2";
export const StyledLayout = (props) => {

  return (
    <React.Fragment>
      <Layout className="layout-background">
        <Header style={{ position: "fixed", zIndex: 1, width: "100%" }} className="nav-temp">
          {/* <div className="logo" />
            <Menu theme="dark" mode="horizontal" defaultSelectedKeys={['2']}>
              <Menu.Item key="1">nav 1</Menu.Item>
              <Menu.Item key="2">nav 2</Menu.Item>
              <Menu.Item key="3">nav 3</Menu.Item>
            </Menu> */}
        </Header>
        <Content
          className="site-layout"
          style={{ padding: "0 50px", marginTop: 64 }}
        >
          <Breadcrumb style={{ margin: "16px 0" }} className="styled-text">
            <Breadcrumb.Item className="styled-text">Home</Breadcrumb.Item>
            <Breadcrumb.Item className="styled-text">List</Breadcrumb.Item>
            <Breadcrumb.Item className="styled-text">App</Breadcrumb.Item>
          </Breadcrumb>
          <div
            className="site-layout-background"
            style={{ padding: 24, minHeight: 380 }}
          >
            <FetchCard></FetchCard>
            {/* <FetchCard></FetchCard> */}
          </div>
        </Content>
        <Footer style={{ textAlign: "center" }} className="nav-temp styled-text">
          Team 2 | Component
        </Footer>
      </Layout>
    </React.Fragment>
  );
};
