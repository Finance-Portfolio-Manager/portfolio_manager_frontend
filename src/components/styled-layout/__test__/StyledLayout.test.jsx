import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import StyledLayout from './../StyledLayout';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
// const wrapper = shallow(<StyledLayout></StyledLayout>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<StyledLayout></StyledLayout>).toJSON();
    expect(tree).toMatchSnapshot();
})