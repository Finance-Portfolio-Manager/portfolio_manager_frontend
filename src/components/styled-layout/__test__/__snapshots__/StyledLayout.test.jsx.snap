// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<section
  className="ant-layout layout-background"
>
  <header
    className="ant-layout-header nav-temp"
    style={
      Object {
        "position": "fixed",
        "width": "100%",
        "zIndex": 1,
      }
    }
  />
  <main
    className="ant-layout-content site-layout"
    style={
      Object {
        "marginTop": 64,
        "padding": "0 50px",
      }
    }
  >
    <div
      className="ant-breadcrumb styled-text"
      style={
        Object {
          "margin": "16px 0",
        }
      }
    >
      <span>
        <span
          className="styled-text"
        >
          Home
        </span>
        <span
          className="ant-breadcrumb-separator"
        >
          /
        </span>
      </span>
      <span>
        <span
          className="styled-text"
        >
          List
        </span>
        <span
          className="ant-breadcrumb-separator"
        >
          /
        </span>
      </span>
      <span>
        <span
          className="styled-text"
        >
          App
        </span>
        <span
          className="ant-breadcrumb-separator"
        >
          /
        </span>
      </span>
    </div>
    <div
      className="site-layout-background"
      style={
        Object {
          "minHeight": 380,
          "padding": 24,
        }
      }
    >
      <div
        className="ant-row grid-container"
        style={Object {}}
      >
        <div
          className="ant-col"
          style={Object {}}
        >
          <div
            className="site-card-border-less-wrapper"
          >
            <div
              className="ant-card card-body"
              style={
                Object {
                  "width": 300,
                }
              }
            >
              <div
                className="ant-card-body"
                style={Object {}}
              >
                <h1
                  className="card-header"
                >
                  @Username
                </h1>
                <div
                  className="chart-container"
                >
                  <canvas
                    className="zoom"
                    data-testid="canvas"
                    height={150}
                    onClick={[Function]}
                    role="img"
                    width={300}
                  />
                </div>
                <h2
                  className="card-header"
                >
                  $118.99
                </h2>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col"
          style={Object {}}
        >
          <div
            className="site-card-border-less-wrapper"
          >
            <div
              className="ant-card card-body"
              style={
                Object {
                  "width": 300,
                }
              }
            >
              <div
                className="ant-card-body"
                style={Object {}}
              >
                <h1
                  className="card-header"
                >
                  @Username
                </h1>
                <div
                  className="chart-container"
                >
                  <canvas
                    className="zoom"
                    data-testid="canvas"
                    height={150}
                    onClick={[Function]}
                    role="img"
                    width={300}
                  />
                </div>
                <h2
                  className="card-header"
                >
                  $118.99
                </h2>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col"
          style={Object {}}
        >
          <div
            className="site-card-border-less-wrapper"
          >
            <div
              className="ant-card card-body"
              style={
                Object {
                  "width": 300,
                }
              }
            >
              <div
                className="ant-card-body"
                style={Object {}}
              >
                <h1
                  className="card-header"
                >
                  @Username
                </h1>
                <div
                  className="chart-container"
                >
                  <canvas
                    className="zoom"
                    data-testid="canvas"
                    height={150}
                    onClick={[Function]}
                    role="img"
                    width={300}
                  />
                </div>
                <h2
                  className="card-header"
                >
                  $118.99
                </h2>
              </div>
            </div>
          </div>
        </div>
        <div
          className="ant-col"
          style={Object {}}
        >
          <div
            className="site-card-border-less-wrapper"
          >
            <div
              className="ant-card card-body"
              style={
                Object {
                  "width": 300,
                }
              }
            >
              <div
                className="ant-card-body"
                style={Object {}}
              >
                <h1
                  className="card-header"
                >
                  @Username
                </h1>
                <div
                  className="chart-container"
                >
                  <canvas
                    className="zoom"
                    data-testid="canvas"
                    height={150}
                    onClick={[Function]}
                    role="img"
                    width={300}
                  />
                </div>
                <h2
                  className="card-header"
                >
                  $118.99
                </h2>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </main>
  <footer
    className="ant-layout-footer nav-temp styled-text"
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Team 2 | Component
  </footer>
</section>
`;
